src/clj/qbits/jet/client/http.clj

```diff
@@ -356,18 +356,21 @@
   "Asynchronously loops and looks for presence of trailers in the response.
    When available, they are propagated to the trailers channel and other request channels closed.
    Please see https://github.com/eclipse/jetty.project/issues/3842 for details on why we need this."
-  [response-trailers-ch response-body-ch close-request-channels! response]
+  [response-trailers-ch response-body-ch content-provider-promise response]
   (async/go
-    (loop []
-      ;; iterate every second and check if trailers are available
-      (async/<! (async/timeout 1000))
-      (let [trailers (.getTrailers ^HttpResponse response)]
-        (if trailers
-          (do
-            (async/>! response-trailers-ch (util/http-fields->map trailers))
-            (close-request-channels!))
-          (when-not (protocols/closed? response-body-ch)
-            (recur)))))))
+    ;; empirically determine the timeout
+    (let [trailer-interval-ms 2000]
+      (loop []
+        ;; iterate and check if trailers are available
+        (async/<! (async/timeout trailer-interval-ms))
+        (let [trailers (.getTrailers ^HttpResponse response)]
+          (if trailers
+            (do
+              (async/>! response-trailers-ch (util/http-fields->map trailers))
+              ;; close the input stream as server has already responded with a trailer
+              (some-> content-provider-promise (deref 0 nil) (.close)))
```

Comment:
Previously we were closing channels.  I don't follow how the same channels are all closed with this change.

Reply:
Should we mention this workaround in the Jetty issue to better illustrate the problem?

Reply:
Added comments to describe how the change avoids closing an actively streaming body channel.